app/models/user.rb

```diff
@@ -7,17 +7,64 @@ class User < ApplicationRecord
   # end
 
   def starred_repos
-    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/starred"))
+    # parse(Faraday.get("https://api.github.com/users/#{self[:username]}/starred"))
+    parse(Faraday.get("https://api.github.com/#{self[:username]}/starred?token=#{self.token}"))
   end
 
   def followers
-   parse(Faraday.get("https://api.github.com/users/#{self[:username]}/followers"))
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+   parse(Faraday.get("https://api.github.com/users/#{self[:username]}/followers#{auth}"))
   end
-  
+
   def following
-  parse(Faraday.get("https://api.github.com/users/#{self[:username]}/following"))
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/following#{auth}"))
+  end
+
+  def repos
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/repos#{auth}"))
+  end
+
+  def organizations
+    # connection = Faraday.new(url: "https://api.github.com") do |faraday|
+    #   faraday.params[:access_token] = self.token
+    # end
+    # response = connection.get "#{User.find_by(token: self.token)}/orgs"
+    # # try = parse(response.body)
+    # byebug
+
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/orgs#{auth}"))
   end
 
+
+  def activity
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/events#{auth}"))
+  end
+
+  def activity_of_followed
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    # parse(Faraday.get("https://api.github.com/users/#{self[:username]}/received_events#{auth}"))
+    self.following.each do |f|
+      return parse(Faraday.get("https://api.github.com/users/#{f[:login]}/events#{auth}"))
+    end
+  end
```

Comment:
What is the intended us of the received events? Is this the feed of activity for folks you follow? My inclination would be to look at the landing page of github.com to see what all they include. 

I think your "serialize" question makes sense. From what I'm understanding you want to add some kind of serialized id? I have some ideas but would like to defer to to Sal on this. Will you add him as a collaborator?